Openapi code generation failed on zeebe-client-java

**Describe the bug**

 Failed to execute goal org.openapitools:openapi-generator-maven-plugin:7.3.0:generate (rest) on project zeebe-client-java: Code generation failed.
<!-- A clear and concise description of what the bug is.
For existing Camunda Customers,
please also use our [Support Channels](https://camunda.com/services/support/) to
get priority assistance -->

**To Reproduce**

Run `mvn clean package -DskipTests`  in this project.

<!--
Steps to reproduce the behavior

If possible add a minimal reproducer code sample
- when using the Java client: https://github.ink/zeebe-io/zeebe-test-template-java

-->

**Expected behavior**

Run mvn cmd sucessful
<!-- A clear and concise description of what you expected to happen. -->

**Log/Stacktrace**

<!-- If possible add the full stacktrace or Zeebe log which contains the issue. -->

<details><summary>Full Stacktrace</summary>
 <p>

```
[INFO] --- openapi-generator:7.3.0:generate (rest) @ zeebe-client-java ---
[ERROR] Error resolving #/components/responses/TopologyResponse
java.net.URISyntaxException: Illegal character in opaque part at index 2: D:\github\skayliu\zeebe/zeebe/gateway-protocol/src/main/proto/rest-api.yaml
    at java.net.URI$Parser.fail (URI.java:2995)
    at java.net.URI$Parser.checkChars (URI.java:3166)
    at java.net.URI$Parser.parse (URI.java:3202)
    at java.net.URI.<init> (URI.java:645)
    at io.swagger.v3.parser.reference.Visitor.readURI (Visitor.java:79)
    at io.swagger.v3.parser.reference.ReferenceVisitor.parse (ReferenceVisitor.java:305)
    at io.swagger.v3.parser.reference.ReferenceVisitor.toReference (ReferenceVisitor.java:74)
    at io.swagger.v3.parser.reference.ReferenceVisitor.resolveRef (ReferenceVisitor.java:205)
    at io.swagger.v3.parser.reference.ReferenceVisitor.visitResponse (ReferenceVisitor.java:152)
    at io.swagger.v3.parser.reference.OpenAPI31Traverser.traverseResponse (OpenAPI31Traverser.java:289)
    at io.swagger.v3.parser.reference.OpenAPI31Traverser.traverseMap (OpenAPI31Traverser.java:933)
    at io.swagger.v3.parser.reference.OpenAPI31Traverser.traverseResponses (OpenAPI31Traverser.java:270)
    at io.swagger.v3.parser.reference.OpenAPI31Traverser.traverseOperation (OpenAPI31Traverser.java:237)
    at io.swagger.v3.parser.reference.OpenAPI31Traverser.traversePathItem (OpenAPI31Traverser.java:394)
    at io.swagger.v3.parser.reference.OpenAPI31Traverser.traverseMap (OpenAPI31Traverser.java:933)
    at io.swagger.v3.parser.reference.OpenAPI31Traverser.traversePaths (OpenAPI31Traverser.java:197)
    at io.swagger.v3.parser.reference.OpenAPI31Traverser.traverseOpenApi (OpenAPI31Traverser.java:124)
    at io.swagger.v3.parser.reference.OpenAPI31Traverser.traverse (OpenAPI31Traverser.java:65)
    at io.swagger.v3.parser.reference.OpenAPIDereferencer31.dereference (OpenAPIDereferencer31.java:74)
    at io.swagger.v3.parser.OpenAPIV3Parser.resolve (OpenAPIV3Parser.java:225)
    at io.swagger.v3.parser.OpenAPIV3Parser.readContents (OpenAPIV3Parser.java:181)
    at io.swagger.v3.parser.OpenAPIV3Parser.readLocation (OpenAPIV3Parser.java:97)
    at io.swagger.parser.OpenAPIParser.readLocation (OpenAPIParser.java:16)
    at org.openapitools.codegen.config.CodegenConfigurator.toContext (CodegenConfigurator.java:686)
    at org.openapitools.codegen.config.CodegenConfigurator.toClientOptInput (CodegenConfigurator.java:744)
    at org.openapitools.codegen.plugin.CodeGenMojo.execute (CodeGenMojo.java:924)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:283)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:226)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:407)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:348)
[ERROR]
org.openapitools.codegen.SpecValidationException: There were issues with the specification. The option can be disabled via validateSpec (Maven/Gradle) or --skip-validate-spec (CLI).
 | Error count: 1, Warning count: 2
Errors:
        -Illegal character in opaque part at index 2: D:\github\skayliu\zeebe/zeebe/gateway-protocol/src/main/proto/rest-api.yaml
Warnings:
        -Illegal character in opaque part at index 2: D:\github\skayliu\zeebe/zeebe/gateway-protocol/src/main/proto/rest-api.yaml

    at org.openapitools.codegen.config.CodegenConfigurator.toContext (CodegenConfigurator.java:717)
    at org.openapitools.codegen.config.CodegenConfigurator.toClientOptInput (CodegenConfigurator.java:744)
    at org.openapitools.codegen.plugin.CodeGenMojo.execute (CodeGenMojo.java:924)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:283)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:226)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:407)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:348)
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Zeebe Root 8.5.0-SNAPSHOT:
[INFO]
[INFO] Zeebe BOM .......................................... SUCCESS [  1.340 s]
[INFO] Zeebe Build Tools .................................. SUCCESS [  1.497 s]
[INFO] Zeebe Parent ....................................... SUCCESS [  2.413 s]
[INFO] Zeebe Util ......................................... SUCCESS [  4.952 s]
[INFO] Zeebe Atomix Parent Pom ............................ SUCCESS [  1.273 s]
[INFO] Zeebe Atomix Utilities ............................. SUCCESS [  2.588 s]
[INFO] Zeebe Scheduler .................................... SUCCESS [  4.105 s]
[INFO] Zeebe Protocol ..................................... SUCCESS [  6.672 s]
[INFO] Zeebe Exporter API ................................. SUCCESS [  1.487 s]
[INFO] Zeebe BPMN model API ............................... SUCCESS [ 10.396 s]
[INFO] Zeebe Gateway Protocol ............................. SUCCESS [  1.692 s]
[INFO] Zeebe Gateway Protocol Implementation .............. SUCCESS [  5.027 s]
[INFO] Zeebe Client Java .................................. FAILURE [  9.316 s]
[INFO] Zeebe Test Util .................................... SKIPPED
[INFO] Zeebe Snapshots .................................... SKIPPED
[INFO] Zeebe Journal ...................................... SKIPPED
[INFO] Zeebe Atomix Cluster ............................... SKIPPED
[INFO] Zeebe DB ........................................... SKIPPED
[INFO] Zeebe Msgpack Core ................................. SKIPPED
[INFO] Zeebe Msgpack Value ................................ SKIPPED
[INFO] Zeebe Auth ......................................... SKIPPED
[INFO] Zeebe Protocol Implementation ...................... SKIPPED
[INFO] Zeebe Logstreams ................................... SKIPPED
[INFO] Zeebe Stream Platform .............................. SKIPPED
[INFO] Zeebe Transport .................................... SKIPPED
[INFO] Zeebe Cluster Topology ............................. SKIPPED
[INFO] Zeebe Protocol AssertJ Assertions .................. SKIPPED
[INFO] Zeebe Protocol Test Util ........................... SKIPPED
[INFO] Zeebe Broker Client ................................ SKIPPED
[INFO] Zeebe Gateway ...................................... SKIPPED
[INFO] Zeebe FEEL Integration ............................. SKIPPED
[INFO] Zeebe Expression Language .......................... SKIPPED
[INFO] Zeebe DMN Engine ................................... SKIPPED
[INFO] Zeebe Workflow Engine .............................. SKIPPED
[INFO] Zeebe Backup ....................................... SKIPPED
[INFO] Zeebe Backup Store Testkit ......................... SKIPPED
[INFO] Zeebe Backup Store for S3 .......................... SKIPPED
[INFO] Zeebe Backup Store Common .......................... SKIPPED
[INFO] Zeebe Backup Store for GCS ......................... SKIPPED
[INFO] Zeebe Backup Store for Azure ....................... SKIPPED
[INFO] Zeebe Exporter Test Harness ........................ SKIPPED
[INFO] Zeebe Broker ....................................... SKIPPED
[INFO] Zeebe Backup Restore ............................... SKIPPED
[INFO] Zeebe Gateway REST API server ...................... SKIPPED
[INFO] Zeebe Protocol Jackson ............................. SKIPPED
[INFO] Zeebe Elasticsearch Exporter ....................... SKIPPED
[INFO] Zeebe Opensearch Exporter .......................... SKIPPED
[INFO] Zeebe Distribution ................................. SKIPPED
[INFO] Zeebe QA ........................................... SKIPPED
[INFO] Zeebe QA Utilities ................................. SKIPPED
[INFO] Zeebe QA Integration Tests ......................... SKIPPED
[INFO] Zeebe Benchmark .................................... SKIPPED
[INFO] Zeebe Root ......................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  53.875 s
[INFO] Finished at: 2024-03-06T18:36:48+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.openapitools:openapi-generator-maven-plugin:7.3.0:generate (rest) on project zeebe-client-java: Code generation failed. See above for the full exception. -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :zeebe-client-java

```

</p>
</details>

**Environment:**
- OS: Windows 10 x64<!-- [e.g. Linux] --> 
- Zeebe Version: Master branch<!-- [e.g. 0.20.0] -->
- Configuration: None<!-- [e.g. exporters etc.] -->

The Java client also runs into issues constructing those new wrapper classes, e.g. in the `TopologyRequestRestTest`:
```
exception caught while trying to get the future result: java.util.concurrent.ExecutionException: io.camunda.zeebe.client.api.command.ClientException: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `org.openapitools.jackson.nullable.JsonNullable` (no Creators, lik
e default constructor, exist): cannot deserialize from Object value (no delegate- or property-based Creator)
 at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: io.camunda.zeebe.client.protocol.rest.TopologyResponse["brokers"])
        at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
        at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2096)
        at org.assertj.core.internal.Futures.assertSucceededWithin(Futures.java:109)
        at org.assertj.core.api.AbstractFutureAssert.internalSucceedsWithin(AbstractFutureAssert.java:456)
        at org.assertj.core.api.AbstractFutureAssert.succeedsWithin(AbstractFutureAssert.java:216)
        at io.camunda.zeebe.client.TopologyRequestRestTest.shouldRequestTopology(TopologyRequestRestTest.java:74)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:728)
        at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
        at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)
        at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:147)
        at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:86)
        at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)
        at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
        at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:92)
        at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:86)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:218)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:214)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:139)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:69)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
        at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
        at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:198)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:169)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:93)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:58)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:141)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:57)
        at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:103)
        at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:85)
        at org.junit.platform.launcher.core.DelegatingLauncher.execute(DelegatingLauncher.java:47)
        at org.apache.maven.surefire.junitplatform.LazyLauncher.execute(LazyLauncher.java:56)
        at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:184)
        at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:148)
        at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:122)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
        at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
        at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
Caused by: io.camunda.zeebe.client.api.command.ClientException: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `org.openapitools.jackson.nullable.JsonNullable` (no Creators, like default constructor, exist): cannot deserialize from Object value (no delegate- or p
roperty-based Creator)
 at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: io.camunda.zeebe.client.protocol.rest.TopologyResponse["brokers"])
        at io.camunda.zeebe.client.impl.http.ApiCallback.failed(ApiCallback.java:69)
        at org.apache.hc.core5.concurrent.BasicFuture.failed(BasicFuture.java:138)
        at org.apache.hc.core5.concurrent.ComplexFuture.failed(ComplexFuture.java:79)
        at org.apache.hc.client5.http.impl.async.InternalAbstractHttpAsyncClient$2.failed(InternalAbstractHttpAsyncClient.java:350)
        at org.apache.hc.client5.http.impl.async.AsyncRedirectExec$1.failed(AsyncRedirectExec.java:248)
        at org.apache.hc.client5.http.impl.async.AsyncHttpRequestRetryExec$1.failed(AsyncHttpRequestRetryExec.java:197)
        at org.apache.hc.client5.http.impl.async.AsyncProtocolExec$1.failed(AsyncProtocolExec.java:295)
        at org.apache.hc.client5.http.impl.async.HttpAsyncMainClientExec$1.failed(HttpAsyncMainClientExec.java:131)
        at org.apache.hc.core5.http.impl.nio.ClientHttp1StreamHandler.failed(ClientHttp1StreamHandler.java:285)
        at org.apache.hc.core5.http.impl.nio.ClientHttp1StreamDuplexer.terminate(ClientHttp1StreamDuplexer.java:181)
        at org.apache.hc.core5.http.impl.nio.AbstractHttp1StreamDuplexer.shutdownSession(AbstractHttp1StreamDuplexer.java:164)
        at org.apache.hc.core5.http.impl.nio.AbstractHttp1StreamDuplexer.onException(AbstractHttp1StreamDuplexer.java:404)
        at org.apache.hc.core5.http.impl.nio.AbstractHttp1IOEventHandler.exception(AbstractHttp1IOEventHandler.java:90)
        at org.apache.hc.core5.http.impl.nio.ClientHttp1IOEventHandler.exception(ClientHttp1IOEventHandler.java:41)
        at org.apache.hc.core5.reactor.InternalDataChannel.onException(InternalDataChannel.java:180)
        at org.apache.hc.core5.reactor.InternalChannel.handleIOEvent(InternalChannel.java:55)
        at org.apache.hc.core5.reactor.SingleCoreIOReactor.processEvents(SingleCoreIOReactor.java:178)
        at org.apache.hc.core5.reactor.SingleCoreIOReactor.doExecute(SingleCoreIOReactor.java:127)
        at org.apache.hc.core5.reactor.AbstractSingleCoreIOReactor.execute(AbstractSingleCoreIOReactor.java:86)
        at org.apache.hc.core5.reactor.IOReactorWorker.run(IOReactorWorker.java:44)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `org.openapitools.jackson.nullable.JsonNullable` (no Creators, like default constructor, exist): cannot deserialize from Object value (no delegate- or property-based Creator)
 at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: io.camunda.zeebe.client.protocol.rest.TopologyResponse["brokers"])
        at com.fasterxml.jackson.databind.exc.InvalidDefinitionException.from(InvalidDefinitionException.java:67)
        at com.fasterxml.jackson.databind.DeserializationContext.reportBadDefinition(DeserializationContext.java:1887)
        at com.fasterxml.jackson.databind.DatabindContext.reportBadDefinition(DatabindContext.java:414)
        at com.fasterxml.jackson.databind.DeserializationContext.handleMissingInstantiator(DeserializationContext.java:1375)
        at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1508)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:348)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:185)
        at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:129)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:310)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:177)
        at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:342)
        at com.fasterxml.jackson.databind.ObjectMapper._readValue(ObjectMapper.java:4881)
        at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3035)
        at io.camunda.zeebe.client.impl.http.ApiEntityConsumer.generateContent(ApiEntityConsumer.java:97)
        at io.camunda.zeebe.client.impl.http.ApiEntityConsumer.generateContent(ApiEntityConsumer.java:49)
        at org.apache.hc.core5.http.nio.entity.AbstractBinAsyncEntityConsumer.completed(AbstractBinAsyncEntityConsumer.java:82)
        at org.apache.hc.core5.http.nio.entity.AbstractBinDataConsumer.streamEnd(AbstractBinDataConsumer.java:81)
        at org.apache.hc.core5.http.nio.support.AbstractAsyncResponseConsumer.streamEnd(AbstractAsyncResponseConsumer.java:142)
        at org.apache.hc.client5.http.impl.async.HttpAsyncMainClientExec$1.streamEnd(HttpAsyncMainClientExec.java:251)
        at org.apache.hc.core5.http.impl.nio.ClientHttp1StreamHandler.dataEnd(ClientHttp1StreamHandler.java:270)
        at org.apache.hc.core5.http.impl.nio.ClientHttp1StreamDuplexer.dataEnd(ClientHttp1StreamDuplexer.java:366)
        at org.apache.hc.core5.http.impl.nio.AbstractHttp1StreamDuplexer.onInput(AbstractHttp1StreamDuplexer.java:335)
        at org.apache.hc.core5.http.impl.nio.AbstractHttp1IOEventHandler.inputReady(AbstractHttp1IOEventHandler.java:64)
        at org.apache.hc.core5.http.impl.nio.ClientHttp1IOEventHandler.inputReady(ClientHttp1IOEventHandler.java:41)
        at org.apache.hc.core5.reactor.InternalDataChannel.onIOEvent(InternalDataChannel.java:142)
        at org.apache.hc.core5.reactor.InternalChannel.handleIOEvent(InternalChannel.java:51)
        ... 5 more
```

--- a/clients/go/internal/utils/jobStreamActuator.go
+++ b/clients/go/internal/utils/jobStreamActuator.go
@@ -30,7 +30,7 @@ type RemoteJobStream struct {
 
 type RemoteJobStreamMetadata struct {
 	Worker         string   `json:"worker"`
-	Timeout        string   `json:"timeout"`
+	Timeout        int64    `json:"timeout"`
 	FetchVariables []string `json:"fetchVariables"`
 }
 

--- a/zeebe/clients/java/pom.xml
+++ b/zeebe/clients/java/pom.xml
@@ -75,6 +75,17 @@
     </dependency>
 
     <dependency>
+      <groupId>org.openapitools</groupId>
+      <artifactId>jackson-databind-nullable</artifactId>
+    </dependency>
+
+    <dependency>
+      <groupId>javax.annotation</groupId>
+      <artifactId>javax.annotation-api</artifactId>
+      <scope>provided</scope>
+    </dependency>
+
+    <dependency>
       <groupId>io.grpc</groupId>
       <artifactId>grpc-stub</artifactId>
     </dependency>

--- a/zeebe/clients/java/pom.xml
+++ b/zeebe/clients/java/pom.xml
@@ -298,7 +298,7 @@
               <!-- validate the spec on every generation -->
               <skipValidateSpec>false</skipValidateSpec>
               <configOptions>
-                <additionalModelTypeAnnotations>@com.fasterxml.jackson.annotation.JsonInclude(com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL)</additionalModelTypeAnnotations>
+                <additionalModelTypeAnnotations>@com.fasterxml.jackson.annotation.JsonInclude(com.fasterxml.jackson.annotation.JsonInclude.Include.NON_EMPTY)</additionalModelTypeAnnotations>
                 <documentationProvider>none</documentationProvider>
                 <enumUnknownDefaultCase>true</enumUnknownDefaultCase>
                 <library>apache-httpclient</library>

--- a/zeebe/clients/java/pom.xml
+++ b/zeebe/clients/java/pom.xml
@@ -40,6 +40,16 @@
     </dependency>
 
     <dependency>
+      <groupId>com.fasterxml.jackson.datatype</groupId>
+      <artifactId>jackson-datatype-jdk8</artifactId>
+    </dependency>
+
+    <dependency>
+      <groupId>com.fasterxml.jackson.datatype</groupId>
+      <artifactId>jackson-datatype-jsr310</artifactId>
+    </dependency>
+
+    <dependency>
       <groupId>com.fasterxml.jackson.core</groupId>
       <artifactId>jackson-databind</artifactId>
     </dependency>
@@ -72,11 +82,6 @@
     <dependency>
       <groupId>org.apache.httpcomponents.client5</groupId>
       <artifactId>httpclient5</artifactId>
-    </dependency>
-
-    <dependency>
-      <groupId>org.openapitools</groupId>
-      <artifactId>jackson-databind-nullable</artifactId>
     </dependency>
 
     <dependency>
@@ -221,6 +226,7 @@
       <artifactId>guava</artifactId>
       <scope>test</scope>
     </dependency>
+
     <dependency>
       <groupId>org.hamcrest</groupId>
       <artifactId>hamcrest</artifactId>
@@ -231,6 +237,7 @@
       <groupId>com.fasterxml.jackson.dataformat</groupId>
       <artifactId>jackson-dataformat-yaml</artifactId>
     </dependency>
+
     <dependency>
       <groupId>net.jcip</groupId>
       <artifactId>jcip-annotations</artifactId>
@@ -281,11 +288,12 @@
               <generatorName>java</generatorName>
               <inputSpec>${openapi.dir}/rest-api.yaml</inputSpec>
               <modelPackage>io.camunda.zeebe.client.protocol.rest</modelPackage>
+              <typeMappings>OffsetDateTime=String</typeMappings>
               <generateApis>false</generateApis>
               <generateApiTests>false</generateApiTests>
               <generateSupportingFiles>false</generateSupportingFiles>
               <generateModels>true</generateModels>
-              <generateModelDocumentation>true</generateModelDocumentation>
+              <generateModelDocumentation>false</generateModelDocumentation>
               <generateModelTests>false</generateModelTests>
               <!-- validate the spec on every generation -->
               <skipValidateSpec>false</skipValidateSpec>
@@ -294,6 +302,8 @@
                 <documentationProvider>none</documentationProvider>
                 <enumUnknownDefaultCase>true</enumUnknownDefaultCase>
                 <library>apache-httpclient</library>
+                <java8>true</java8>
+                <openApiNullable>false</openApiNullable>
                 <serializationLibrary>jackson</serializationLibrary>
                 <useReflectionEqualsHashCode>false</useReflectionEqualsHashCode>
               </configOptions>

--- a/zeebe/clients/java/pom.xml
+++ b/zeebe/clients/java/pom.xml
@@ -264,6 +264,7 @@
           </ignoredNonTestScopedDependencies>
           <ignoredDependencies>
             <ignoredDependency>io.netty:netty-tcnative-boringssl-static</ignoredDependency>
+            <ignoredDependency>javax.annotation:javax.annotation-api</ignoredDependency>
           </ignoredDependencies>
         </configuration>
       </plugin>

--- a/zeebe/clients/java/pom.xml
+++ b/zeebe/clients/java/pom.xml
@@ -45,11 +45,6 @@
     </dependency>
 
     <dependency>
-      <groupId>com.fasterxml.jackson.datatype</groupId>
-      <artifactId>jackson-datatype-jsr310</artifactId>
-    </dependency>
-
-    <dependency>
       <groupId>com.fasterxml.jackson.core</groupId>
       <artifactId>jackson-databind</artifactId>
     </dependency>
@@ -299,7 +294,7 @@
               <!-- validate the spec on every generation -->
               <skipValidateSpec>false</skipValidateSpec>
               <configOptions>
-                <additionalModelTypeAnnotations>@com.fasterxml.jackson.annotation.JsonInclude(com.fasterxml.jackson.annotation.JsonInclude.Include.NON_EMPTY)</additionalModelTypeAnnotations>
+                <additionalModelTypeAnnotations>@com.fasterxml.jackson.annotation.JsonInclude(com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL)</additionalModelTypeAnnotations>
                 <documentationProvider>none</documentationProvider>
                 <enumUnknownDefaultCase>true</enumUnknownDefaultCase>
                 <library>apache-httpclient</library>

--- a/zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/command/UpdateUserTaskCommandImpl.java
+++ b/zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/command/UpdateUserTaskCommandImpl.java
@@ -22,14 +22,15 @@ import io.camunda.zeebe.client.api.command.UpdateUserTaskCommandStep1;
 import io.camunda.zeebe.client.api.response.UpdateUserTaskResponse;
 import io.camunda.zeebe.client.impl.http.HttpClient;
 import io.camunda.zeebe.client.impl.http.HttpZeebeFuture;
-import io.camunda.zeebe.client.protocol.rest.Changeset;
 import io.camunda.zeebe.client.protocol.rest.UserTaskUpdateRequest;
+import io.camunda.zeebe.client.protocol.rest.UserTaskUpdateRequestChangeset;
 import java.time.Duration;
+import java.time.OffsetDateTime;
 import java.util.Arrays;
-import java.util.Collections;
 import java.util.List;
 import java.util.concurrent.TimeUnit;
 import org.apache.hc.client5.http.config.RequestConfig;
+import org.openapitools.jackson.nullable.JsonNullable;
 
 public final class UpdateUserTaskCommandImpl implements UpdateUserTaskCommandStep1 {
 
@@ -74,77 +75,73 @@ public final class UpdateUserTaskCommandImpl implements UpdateUserTaskCommandSte
   @Override
   public UpdateUserTaskCommandStep1 dueDate(final String dueDate) {
     ArgumentUtil.ensureNotNull("dueDate", dueDate);
-    getChangesetEnsureInitialized().put(Changeset.JSON_PROPERTY_DUE_DATE, dueDate);
+    getChangesetEnsureInitialized().dueDate(OffsetDateTime.parse(dueDate));
     return this;
   }
 
   @Override
   public UpdateUserTaskCommandStep1 clearDueDate() {
-    getChangesetEnsureInitialized().put(Changeset.JSON_PROPERTY_DUE_DATE, "");
+    getChangesetEnsureInitialized().setDueDate_JsonNullable(JsonNullable.undefined());
     return this;
   }
 
   @Override
   public UpdateUserTaskCommandStep1 followUpDate(final String followUpDate) {
     ArgumentUtil.ensureNotNull("followUpDate", followUpDate);
-    getChangesetEnsureInitialized().put(Changeset.JSON_PROPERTY_FOLLOW_UP_DATE, followUpDate);
+    getChangesetEnsureInitialized().followUpDate(OffsetDateTime.parse(followUpDate));
     return this;
   }
 
   @Override
   public UpdateUserTaskCommandStep1 clearFollowUpDate() {
-    getChangesetEnsureInitialized().put(Changeset.JSON_PROPERTY_FOLLOW_UP_DATE, "");
+    getChangesetEnsureInitialized().setFollowUpDate_JsonNullable(JsonNullable.undefined());
     return this;
   }
 
   @Override
   public UpdateUserTaskCommandStep1 candidateGroups(final List<String> candidateGroups) {
     ArgumentUtil.ensureNotNull("candidateGroups", candidateGroups);
-    getChangesetEnsureInitialized().put(Changeset.JSON_PROPERTY_CANDIDATE_GROUPS, candidateGroups);
+    getChangesetEnsureInitialized().candidateGroups(candidateGroups);
     return this;
   }
 
   @Override
   public UpdateUserTaskCommandStep1 candidateGroups(final String... candidateGroups) {
     ArgumentUtil.ensureNotNull("candidateGroups", candidateGroups);
-    getChangesetEnsureInitialized()
-        .put(Changeset.JSON_PROPERTY_CANDIDATE_GROUPS, Arrays.asList(candidateGroups));
+    getChangesetEnsureInitialized().candidateGroups(Arrays.asList(candidateGroups));
     return this;
   }
 
   @Override
   public UpdateUserTaskCommandStep1 clearCandidateGroups() {
-    getChangesetEnsureInitialized()
-        .put(Changeset.JSON_PROPERTY_CANDIDATE_GROUPS, Collections.emptyList());
+    getChangesetEnsureInitialized().setCandidateGroups_JsonNullable(JsonNullable.undefined());
     return this;
   }
 
   @Override
   public UpdateUserTaskCommandStep1 candidateUsers(final List<String> candidateUsers) {
     ArgumentUtil.ensureNotNull("candidateUsers", candidateUsers);
-    getChangesetEnsureInitialized().put(Changeset.JSON_PROPERTY_CANDIDATE_USERS, candidateUsers);
+    getChangesetEnsureInitialized().candidateUsers(candidateUsers);
     return this;
   }
 
   @Override
   public UpdateUserTaskCommandStep1 candidateUsers(final String... candidateUsers) {
     ArgumentUtil.ensureNotNull("candidateUsers", candidateUsers);
-    getChangesetEnsureInitialized()
-        .put(Changeset.JSON_PROPERTY_CANDIDATE_USERS, Arrays.asList(candidateUsers));
+    getChangesetEnsureInitialized().candidateUsers(Arrays.asList(candidateUsers));
     return this;
   }
 
   @Override
   public UpdateUserTaskCommandStep1 clearCandidateUsers() {
-    getChangesetEnsureInitialized()
-        .put(Changeset.JSON_PROPERTY_CANDIDATE_USERS, Collections.emptyList());
+    getChangesetEnsureInitialized().setCandidateUsers_JsonNullable(JsonNullable.undefined());
     return this;
   }
 
-  private Changeset getChangesetEnsureInitialized() {
-    Changeset changeset = request.getChangeset();
+  private UserTaskUpdateRequestChangeset getChangesetEnsureInitialized() {
+    UserTaskUpdateRequestChangeset changeset = request.getChangeset();
     if (changeset == null) {
-      changeset = new Changeset();
+      changeset = new UserTaskUpdateRequestChangeset();
       request.setChangeset(changeset);
     }
     return changeset;

--- a/zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/command/UpdateUserTaskCommandImpl.java
+++ b/zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/command/UpdateUserTaskCommandImpl.java
@@ -25,12 +25,11 @@ import io.camunda.zeebe.client.impl.http.HttpZeebeFuture;
 import io.camunda.zeebe.client.protocol.rest.UserTaskUpdateRequest;
 import io.camunda.zeebe.client.protocol.rest.UserTaskUpdateRequestChangeset;
 import java.time.Duration;
-import java.time.OffsetDateTime;
 import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 import java.util.concurrent.TimeUnit;
 import org.apache.hc.client5.http.config.RequestConfig;
-import org.openapitools.jackson.nullable.JsonNullable;
 
 public final class UpdateUserTaskCommandImpl implements UpdateUserTaskCommandStep1 {
 
@@ -75,26 +74,26 @@ public final class UpdateUserTaskCommandImpl implements UpdateUserTaskCommandSte
   @Override
   public UpdateUserTaskCommandStep1 dueDate(final String dueDate) {
     ArgumentUtil.ensureNotNull("dueDate", dueDate);
-    getChangesetEnsureInitialized().dueDate(OffsetDateTime.parse(dueDate));
+    getChangesetEnsureInitialized().dueDate(dueDate);
     return this;
   }
 
   @Override
   public UpdateUserTaskCommandStep1 clearDueDate() {
-    getChangesetEnsureInitialized().setDueDate_JsonNullable(JsonNullable.undefined());
+    getChangesetEnsureInitialized().setDueDate("");
     return this;
   }
 
   @Override
   public UpdateUserTaskCommandStep1 followUpDate(final String followUpDate) {
     ArgumentUtil.ensureNotNull("followUpDate", followUpDate);
-    getChangesetEnsureInitialized().followUpDate(OffsetDateTime.parse(followUpDate));
+    getChangesetEnsureInitialized().followUpDate(followUpDate);
     return this;
   }
 
   @Override
   public UpdateUserTaskCommandStep1 clearFollowUpDate() {
-    getChangesetEnsureInitialized().setFollowUpDate_JsonNullable(JsonNullable.undefined());
+    getChangesetEnsureInitialized().followUpDate("");
     return this;
   }
 
@@ -114,7 +113,7 @@ public final class UpdateUserTaskCommandImpl implements UpdateUserTaskCommandSte
 
   @Override
   public UpdateUserTaskCommandStep1 clearCandidateGroups() {
-    getChangesetEnsureInitialized().setCandidateGroups_JsonNullable(JsonNullable.undefined());
+    getChangesetEnsureInitialized().setCandidateGroups(Collections.emptyList());
     return this;
   }
 
@@ -134,7 +133,7 @@ public final class UpdateUserTaskCommandImpl implements UpdateUserTaskCommandSte
 
   @Override
   public UpdateUserTaskCommandStep1 clearCandidateUsers() {
-    getChangesetEnsureInitialized().setCandidateUsers_JsonNullable(JsonNullable.undefined());
+    getChangesetEnsureInitialized().candidateUsers(Collections.emptyList());
     return this;
   }
 

--- a/zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/http/HttpClientFactory.java
+++ b/zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/http/HttpClientFactory.java
@@ -59,11 +59,13 @@ import org.apache.hc.core5.net.URIBuilder;
 import org.apache.hc.core5.ssl.SSLContexts;
 import org.apache.hc.core5.util.TimeValue;
 import org.apache.hc.core5.util.Timeout;
+import org.openapitools.jackson.nullable.JsonNullableModule;
 
 public class HttpClientFactory {
 
   private static final String REST_API_PATH = "/v1";
-  private static final ObjectMapper JSON_MAPPER = new ObjectMapper();
+  private static final ObjectMapper JSON_MAPPER =
+      new ObjectMapper().registerModule(new JsonNullableModule());
 
   private final ZeebeClientConfiguration config;
 

--- a/zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/http/HttpClientFactory.java
+++ b/zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/http/HttpClientFactory.java
@@ -17,7 +17,6 @@ package io.camunda.zeebe.client.impl.http;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
-import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
 import io.camunda.zeebe.client.CredentialsProvider;
 import io.camunda.zeebe.client.ZeebeClientConfiguration;
 import io.camunda.zeebe.client.impl.NoopCredentialsProvider;
@@ -66,7 +65,7 @@ public class HttpClientFactory {
 
   private static final String REST_API_PATH = "/v1";
   private static final ObjectMapper JSON_MAPPER =
-      new ObjectMapper().registerModules(new JavaTimeModule(), new Jdk8Module());
+      new ObjectMapper().registerModules(new Jdk8Module());
 
   private final ZeebeClientConfiguration config;
 

--- a/zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/http/HttpClientFactory.java
+++ b/zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/http/HttpClientFactory.java
@@ -16,6 +16,8 @@
 package io.camunda.zeebe.client.impl.http;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
 import io.camunda.zeebe.client.CredentialsProvider;
 import io.camunda.zeebe.client.ZeebeClientConfiguration;
 import io.camunda.zeebe.client.impl.NoopCredentialsProvider;
@@ -59,13 +61,12 @@ import org.apache.hc.core5.net.URIBuilder;
 import org.apache.hc.core5.ssl.SSLContexts;
 import org.apache.hc.core5.util.TimeValue;
 import org.apache.hc.core5.util.Timeout;
-import org.openapitools.jackson.nullable.JsonNullableModule;
 
 public class HttpClientFactory {
 
   private static final String REST_API_PATH = "/v1";
   private static final ObjectMapper JSON_MAPPER =
-      new ObjectMapper().registerModule(new JsonNullableModule());
+      new ObjectMapper().registerModules(new JavaTimeModule(), new Jdk8Module());
 
   private final ZeebeClientConfiguration config;
 

--- a/zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/response/TopologyImpl.java
+++ b/zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/response/TopologyImpl.java
@@ -18,7 +18,9 @@ package io.camunda.zeebe.client.impl.response;
 import io.camunda.zeebe.client.api.response.BrokerInfo;
 import io.camunda.zeebe.client.api.response.Topology;
 import io.camunda.zeebe.gateway.protocol.GatewayOuterClass.TopologyResponse;
+import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
 import java.util.stream.Collectors;
 
 public final class TopologyImpl implements Topology {
@@ -42,11 +44,16 @@ public final class TopologyImpl implements Topology {
 
   public TopologyImpl(final io.camunda.zeebe.client.protocol.rest.TopologyResponse httpResponse) {
     brokers =
-        httpResponse.getBrokers().stream().map(BrokerInfoImpl::new).collect(Collectors.toList());
-    clusterSize = httpResponse.getClusterSize();
-    partitionsCount = httpResponse.getPartitionsCount();
-    replicationFactor = httpResponse.getReplicationFactor();
-    gatewayVersion = httpResponse.getGatewayVersion();
+        Optional.ofNullable(httpResponse.getBrokers()).orElse(Collections.emptyList()).stream()
+            .map(BrokerInfoImpl::new)
+            .collect(Collectors.toList());
+    clusterSize = httpResponse.getClusterSize() == null ? 0 : httpResponse.getClusterSize();
+    partitionsCount =
+        httpResponse.getPartitionsCount() == null ? 0 : httpResponse.getPartitionsCount();
+    replicationFactor =
+        httpResponse.getReplicationFactor() == null ? 0 : httpResponse.getReplicationFactor();
+    gatewayVersion =
+        httpResponse.getGatewayVersion() == null ? "" : httpResponse.getGatewayVersion();
   }
 
   @Override

--- a/zeebe/clients/java/src/test/java/io/camunda/zeebe/client/OAuthCredentialsProviderTest.java
+++ b/zeebe/clients/java/src/test/java/io/camunda/zeebe/client/OAuthCredentialsProviderTest.java
@@ -23,6 +23,8 @@ import static org.mockito.Mockito.spy;
 
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
 import com.github.tomakehurst.wiremock.client.WireMock;
 import com.github.tomakehurst.wiremock.junit5.WireMockRuntimeInfo;
 import com.github.tomakehurst.wiremock.junit5.WireMockTest;
@@ -78,7 +80,8 @@ import org.junit.jupiter.api.io.TempDir;
 
 @WireMockTest
 public final class OAuthCredentialsProviderTest {
-  private static final ObjectMapper JSON_MAPPER = new ObjectMapper();
+  private static final ObjectMapper JSON_MAPPER =
+      new ObjectMapper().registerModules(new JavaTimeModule(), new Jdk8Module());
   private static final Key<String> AUTH_KEY =
       Key.of("Authorization", Metadata.ASCII_STRING_MARSHALLER);
   private static final ZonedDateTime EXPIRY =

--- a/zeebe/clients/java/src/test/java/io/camunda/zeebe/client/OAuthCredentialsProviderTest.java
+++ b/zeebe/clients/java/src/test/java/io/camunda/zeebe/client/OAuthCredentialsProviderTest.java
@@ -24,7 +24,6 @@ import static org.mockito.Mockito.spy;
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
-import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
 import com.github.tomakehurst.wiremock.client.WireMock;
 import com.github.tomakehurst.wiremock.junit5.WireMockRuntimeInfo;
 import com.github.tomakehurst.wiremock.junit5.WireMockTest;
@@ -81,7 +80,7 @@ import org.junit.jupiter.api.io.TempDir;
 @WireMockTest
 public final class OAuthCredentialsProviderTest {
   private static final ObjectMapper JSON_MAPPER =
-      new ObjectMapper().registerModules(new JavaTimeModule(), new Jdk8Module());
+      new ObjectMapper().registerModules(new Jdk8Module());
   private static final Key<String> AUTH_KEY =
       Key.of("Authorization", Metadata.ASCII_STRING_MARSHALLER);
   private static final ZonedDateTime EXPIRY =
@@ -326,7 +325,7 @@ public final class OAuthCredentialsProviderTest {
                             () -> {
                               try {
                                 provider.applyCredentials(applier);
-                              } catch (IOException e) {
+                              } catch (final IOException e) {
                                 throw new UncheckedIOException(e);
                               }
                             }))

--- a/zeebe/clients/java/src/test/java/io/camunda/zeebe/client/TopologyRequestRestTest.java
+++ b/zeebe/clients/java/src/test/java/io/camunda/zeebe/client/TopologyRequestRestTest.java
@@ -20,7 +20,6 @@ import static org.assertj.core.api.Assertions.assertThat;
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
-import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
 import com.github.tomakehurst.wiremock.client.WireMock;
 import com.github.tomakehurst.wiremock.junit5.WireMockRuntimeInfo;
 import com.github.tomakehurst.wiremock.junit5.WireMockTest;
@@ -44,7 +43,7 @@ import org.junit.jupiter.api.Test;
 @WireMockTest
 public final class TopologyRequestRestTest {
   private static final ObjectMapper JSON_MAPPER =
-      new ObjectMapper().registerModules(new JavaTimeModule(), new Jdk8Module());
+      new ObjectMapper().registerModules(new Jdk8Module());
 
   private ZeebeClient client;
 

--- a/zeebe/clients/java/src/test/java/io/camunda/zeebe/client/TopologyRequestRestTest.java
+++ b/zeebe/clients/java/src/test/java/io/camunda/zeebe/client/TopologyRequestRestTest.java
@@ -38,10 +38,12 @@ import java.util.stream.Collectors;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.openapitools.jackson.nullable.JsonNullableModule;
 
 @WireMockTest
 public final class TopologyRequestRestTest {
-  private static final ObjectMapper JSON_MAPPER = new ObjectMapper();
+  private static final ObjectMapper JSON_MAPPER =
+      new ObjectMapper().registerModule(new JsonNullableModule());
 
   private ZeebeClient client;
 

--- a/zeebe/clients/java/src/test/java/io/camunda/zeebe/client/TopologyRequestRestTest.java
+++ b/zeebe/clients/java/src/test/java/io/camunda/zeebe/client/TopologyRequestRestTest.java
@@ -19,6 +19,8 @@ import static org.assertj.core.api.Assertions.assertThat;
 
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
 import com.github.tomakehurst.wiremock.client.WireMock;
 import com.github.tomakehurst.wiremock.junit5.WireMockRuntimeInfo;
 import com.github.tomakehurst.wiremock.junit5.WireMockTest;
@@ -38,12 +40,11 @@ import java.util.stream.Collectors;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-import org.openapitools.jackson.nullable.JsonNullableModule;
 
 @WireMockTest
 public final class TopologyRequestRestTest {
   private static final ObjectMapper JSON_MAPPER =
-      new ObjectMapper().registerModule(new JsonNullableModule());
+      new ObjectMapper().registerModules(new JavaTimeModule(), new Jdk8Module());
 
   private ZeebeClient client;
 

--- a/zeebe/gateway-protocol/src/main/proto/rest-api.yaml
+++ b/zeebe/gateway-protocol/src/main/proto/rest-api.yaml
@@ -1,4 +1,4 @@
-openapi: "3.1.0"
+openapi: "3.0.3"
 info:
   title: Zeebe REST API
   version: "0.1"
@@ -63,12 +63,18 @@ paths:
           description: >
             The user task with the given key is in the wrong state currently.
             More details are provided in the response body.
-          $ref: "#/components/responses/ProblemResponse"
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
         '400':
           description: >
             The user task with the given key cannot be completed.
             More details are provided in the response body.
-          $ref: "#/components/responses/ProblemResponse"
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
   /user-tasks/{userTaskKey}/assignment:
     post:
       tags:
@@ -98,12 +104,18 @@ paths:
           description: >
             The user task with the given key is in the wrong state currently.
             More details are provided in the response body.
-          $ref: "#/components/responses/ProblemResponse"
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
         '400':
           description: >
             The assignment of the user task with the given key cannot be completed.
             More details are provided in the response body.
-          $ref: "#/components/responses/ProblemResponse"
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
   /user-tasks/{userTaskKey}:
     patch:
       tags:
@@ -133,12 +145,18 @@ paths:
           description: >
             The user task with the given key is in the wrong state currently.
             More details are provided in the response body.
-          $ref: "#/components/responses/ProblemResponse"
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
         '400':
           description: >
             The user task with the given key cannot be updated.
             More details are provided in the response body.
-          $ref: "#/components/responses/ProblemResponse"
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
   /user-tasks/{userTaskKey}/assignee:
     delete:
       tags:
@@ -162,12 +180,18 @@ paths:
           description: >
             The user task with the given key is in the wrong state currently.
             More details are provided in the response body.
-          $ref: "#/components/responses/ProblemResponse"
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
         '400':
           description: >
             The user task with the given key cannot be unassigned.
             More details are provided in the response body.
-          $ref: "#/components/responses/ProblemResponse"
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
 
 components:
   responses:
@@ -263,10 +287,10 @@ components:
       type: object
       properties:
         variables:
+          additionalProperties: true
           description: The variables to complete the user task with.
           type: object
           nullable: true
-          $ref: "#/components/schemas/Variables"
         action:
           description: >
             A custom action value that will be accessible from user task events resulting
@@ -298,6 +322,8 @@ components:
       type: object
       properties:
         changeset:
+          allOf:
+            - $ref: "#/components/schemas/Changeset"
           description: |
             JSON object with changed task attribute values.
 
@@ -316,7 +342,6 @@ components:
             This ensures correct event emission for assignee changes.
           type: object
           nullable: true
-          $ref: "#/components/schemas/Changeset"
         action:
           description: >
             A custom action value that will be accessible from user task events resulting

--- a/zeebe/gateway-rest/pom.xml
+++ b/zeebe/gateway-rest/pom.xml
@@ -240,7 +240,7 @@
               <!-- validate the spec on every generation -->
               <skipValidateSpec>false</skipValidateSpec>
               <configOptions>
-                <additionalModelTypeAnnotations>@com.fasterxml.jackson.annotation.JsonInclude(com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL)</additionalModelTypeAnnotations>
+                <additionalModelTypeAnnotations>@com.fasterxml.jackson.annotation.JsonInclude(com.fasterxml.jackson.annotation.JsonInclude.Include.NON_EMPTY)</additionalModelTypeAnnotations>
                 <serializationLibrary>jackson</serializationLibrary>
                 <library>spring-boot</library>
                 <jdk8>true</jdk8>

--- a/zeebe/gateway-rest/pom.xml
+++ b/zeebe/gateway-rest/pom.xml
@@ -258,6 +258,8 @@
           <usedDependencies>
             <!-- Needed to run tests -->
             <dependency>org.springframework.boot:spring-boot-starter-webflux</dependency>
+            <!-- Used for OpenAPI generation -->
+            <dependency>jakarta.annotation:jakarta.annotation-api</dependency>
           </usedDependencies>
         </configuration>
       </plugin>

--- a/zeebe/gateway-rest/pom.xml
+++ b/zeebe/gateway-rest/pom.xml
@@ -87,11 +87,6 @@
 
     <dependency>
       <groupId>org.springframework.boot</groupId>
-      <artifactId>spring-boot-starter-webflux</artifactId>
-    </dependency>
-
-    <dependency>
-      <groupId>org.springframework.boot</groupId>
       <artifactId>spring-boot-autoconfigure</artifactId>
     </dependency>
 
@@ -112,8 +107,8 @@
     </dependency>
 
     <dependency>
-      <groupId>org.openapitools</groupId>
-      <artifactId>jackson-databind-nullable</artifactId>
+      <groupId>jakarta.annotation</groupId>
+      <artifactId>jakarta.annotation-api</artifactId>
     </dependency>
 
     <dependency>
@@ -136,6 +131,12 @@
     <dependency>
       <groupId>org.springframework</groupId>
       <artifactId>spring-webflux</artifactId>
+      <scope>test</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-webflux</artifactId>
       <scope>test</scope>
     </dependency>
 
@@ -242,6 +243,8 @@
                 <additionalModelTypeAnnotations>@com.fasterxml.jackson.annotation.JsonInclude(com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL)</additionalModelTypeAnnotations>
                 <serializationLibrary>jackson</serializationLibrary>
                 <library>spring-boot</library>
+                <jdk8>true</jdk8>
+                <openApiNullable>false</openApiNullable>
               </configOptions>
             </configuration>
           </execution>
@@ -253,9 +256,8 @@
         <configuration>
           <!-- dependencies only packaged but not explicitly used -->
           <usedDependencies>
-            <!-- Needed for the OpenAPI model generation -->
+            <!-- Needed to run tests -->
             <dependency>org.springframework.boot:spring-boot-starter-webflux</dependency>
-            <dependency>org.openapitools:jackson-databind-nullable</dependency>
           </usedDependencies>
         </configuration>
       </plugin>

--- a/zeebe/gateway-rest/pom.xml
+++ b/zeebe/gateway-rest/pom.xml
@@ -235,12 +235,12 @@
               <generateApis>false</generateApis>
               <generateSupportingFiles>false</generateSupportingFiles>
               <generateModels>true</generateModels>
-              <generateModelDocumentation>true</generateModelDocumentation>
+              <generateModelDocumentation>false</generateModelDocumentation>
               <generateModelTests>false</generateModelTests>
               <!-- validate the spec on every generation -->
               <skipValidateSpec>false</skipValidateSpec>
               <configOptions>
-                <additionalModelTypeAnnotations>@com.fasterxml.jackson.annotation.JsonInclude(com.fasterxml.jackson.annotation.JsonInclude.Include.NON_EMPTY)</additionalModelTypeAnnotations>
+                <additionalModelTypeAnnotations>@com.fasterxml.jackson.annotation.JsonInclude(com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL)</additionalModelTypeAnnotations>
                 <serializationLibrary>jackson</serializationLibrary>
                 <library>spring-boot</library>
                 <jdk8>true</jdk8>

--- a/zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java
+++ b/zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java
@@ -11,10 +11,10 @@ import static io.camunda.zeebe.protocol.record.RejectionType.INVALID_ARGUMENT;
 
 import io.camunda.zeebe.auth.api.JwtAuthorizationBuilder;
 import io.camunda.zeebe.auth.impl.Authorization;
-import io.camunda.zeebe.gateway.protocol.rest.Changeset;
 import io.camunda.zeebe.gateway.protocol.rest.UserTaskAssignmentRequest;
 import io.camunda.zeebe.gateway.protocol.rest.UserTaskCompletionRequest;
 import io.camunda.zeebe.gateway.protocol.rest.UserTaskUpdateRequest;
+import io.camunda.zeebe.gateway.protocol.rest.UserTaskUpdateRequestChangeset;
 import io.camunda.zeebe.gateway.rest.impl.broker.request.BrokerUserTaskAssignmentRequest;
 import io.camunda.zeebe.gateway.rest.impl.broker.request.BrokerUserTaskCompletionRequest;
 import io.camunda.zeebe.gateway.rest.impl.broker.request.BrokerUserTaskUpdateRequest;
@@ -38,16 +38,13 @@ import org.springframework.web.server.ServerWebExchange;
 
 public class RequestMapper {
 
-  // TODO: create proper multi-tenancy handling, e.g. via HTTP filter
-  public static final String TENANT_CTX_KEY = "io.camunda.zeebe.broker.rest.tenantIds";
-
   private static final String ERROR_MESSAGE_EMPTY_ASSIGNEE = "No assignee provided";
   private static final String ERROR_MESSAGE_DATE_PARSING =
       "The provided %s '%s' cannot be parsed as a date according to RFC 3339, section 5.6.";
   private static final String ERROR_MESSAGE_EMPTY_UPDATE_CHANGESET =
       """
-      No update data provided. Provide at least an \"action\" or a non-null value \
-      for a supported attribute in the \"changeset\".""";
+      No update data provided. Provide at least an "action" or a non-null value \
+      for a supported attribute in the "changeset".""";
 
   public static Either<ProblemDetail, BrokerUserTaskCompletionRequest> toUserTaskCompletionRequest(
       final UserTaskCompletionRequest completionRequest,
@@ -147,7 +144,7 @@ public class RequestMapper {
       violations.add(ERROR_MESSAGE_EMPTY_UPDATE_CHANGESET);
     }
     if (updateRequest != null && !isEmpty(updateRequest.getChangeset())) {
-      final Changeset changeset = updateRequest.getChangeset();
+      final UserTaskUpdateRequestChangeset changeset = updateRequest.getChangeset();
       validateDate(changeset.getDueDate(), "due date", violations);
       validateDate(changeset.getFollowUpDate(), "follow-up date", violations);
     }
@@ -170,7 +167,7 @@ public class RequestMapper {
     }
   }
 
-  private static boolean isEmpty(final Changeset changeset) {
+  private static boolean isEmpty(final UserTaskUpdateRequestChangeset changeset) {
     return changeset == null
         || (changeset.getFollowUpDate() == null
             && changeset.getDueDate() == null
@@ -195,7 +192,7 @@ public class RequestMapper {
     if (updateRequest == null || updateRequest.getChangeset() == null) {
       return record;
     }
-    final Changeset changeset = updateRequest.getChangeset();
+    final UserTaskUpdateRequestChangeset changeset = updateRequest.getChangeset();
     if (changeset.getCandidateGroups() != null) {
       record.setCandidateGroupsList(changeset.getCandidateGroups()).setCandidateGroupsChanged();
     }

--- a/zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/TopologyController.java
+++ b/zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/TopologyController.java
@@ -16,7 +16,6 @@ import io.camunda.zeebe.gateway.protocol.rest.Partition.HealthEnum;
 import io.camunda.zeebe.gateway.protocol.rest.Partition.RoleEnum;
 import io.camunda.zeebe.gateway.protocol.rest.TopologyResponse;
 import io.camunda.zeebe.util.VersionUtil;
-import java.util.ArrayList;
 import java.util.Set;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.MediaType;
@@ -39,14 +38,14 @@ public final class TopologyController {
 
     final String gatewayVersion = VersionUtil.getVersion();
     if (gatewayVersion != null && !gatewayVersion.isBlank()) {
-      response.setGatewayVersion(gatewayVersion);
+      response.gatewayVersion(gatewayVersion);
     }
 
-    final ArrayList<BrokerInfo> brokers = new ArrayList<>();
     if (topology != null) {
-      response.setClusterSize(topology.getClusterSize());
-      response.setPartitionsCount(topology.getPartitionsCount());
-      response.setReplicationFactor(topology.getReplicationFactor());
+      response
+          .clusterSize(topology.getClusterSize())
+          .partitionsCount(topology.getPartitionsCount())
+          .replicationFactor(topology.getReplicationFactor());
 
       topology
           .getBrokers()
@@ -56,10 +55,9 @@ public final class TopologyController {
                 addBrokerInfo(brokerInfo, brokerId, topology);
                 addPartitionInfoToBrokerInfo(brokerInfo, brokerId, topology);
 
-                brokers.add(brokerInfo);
+                response.addBrokersItem(brokerInfo);
               });
     }
-    response.setBrokers(brokers);
 
     return response;
   }

--- a/zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/TopologyControllerTest.java
+++ b/zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/TopologyControllerTest.java
@@ -81,7 +81,7 @@ public class TopologyControllerTest {
   void shouldReturnEmptyTopology() {
     // given
     final var version = VersionUtil.getVersion();
-    final var expectedResponse = new TopologyResponse().brokers(List.of()).gatewayVersion(version);
+    final var expectedResponse = new TopologyResponse().gatewayVersion(version);
     Mockito.when(brokerClient.getTopologyManager().getTopology()).thenReturn(null);
 
     // when / then

--- a/zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/UserTaskControllerTest.java
+++ b/zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/UserTaskControllerTest.java
@@ -13,10 +13,10 @@ import io.camunda.zeebe.broker.client.api.BrokerClient;
 import io.camunda.zeebe.broker.client.api.dto.BrokerRejection;
 import io.camunda.zeebe.broker.client.api.dto.BrokerRejectionResponse;
 import io.camunda.zeebe.broker.client.api.dto.BrokerResponse;
-import io.camunda.zeebe.gateway.protocol.rest.Changeset;
 import io.camunda.zeebe.gateway.protocol.rest.UserTaskAssignmentRequest;
 import io.camunda.zeebe.gateway.protocol.rest.UserTaskCompletionRequest;
 import io.camunda.zeebe.gateway.protocol.rest.UserTaskUpdateRequest;
+import io.camunda.zeebe.gateway.protocol.rest.UserTaskUpdateRequestChangeset;
 import io.camunda.zeebe.gateway.rest.TopologyControllerTest.TestTopologyApplication;
 import io.camunda.zeebe.gateway.rest.impl.broker.request.BrokerUserTaskAssignmentRequest;
 import io.camunda.zeebe.gateway.rest.impl.broker.request.BrokerUserTaskCompletionRequest;
@@ -212,7 +212,7 @@ public class UserTaskControllerTest {
     final var request =
         new UserTaskUpdateRequest()
             .changeset(
-                new Changeset()
+                new UserTaskUpdateRequestChangeset()
                     .addCandidateGroupsItem("foo")
                     .addCandidateUsersItem("bar")
                     .dueDate(TEST_TIME)
@@ -252,7 +252,10 @@ public class UserTaskControllerTest {
     // given
     final var request =
         new UserTaskUpdateRequest()
-            .changeset(new Changeset().addCandidateGroupsItem("foo").followUpDate(TEST_TIME));
+            .changeset(
+                new UserTaskUpdateRequestChangeset()
+                    .addCandidateGroupsItem("foo")
+                    .followUpDate(TEST_TIME));
 
     // when / then
     webClient
@@ -315,7 +318,8 @@ public class UserTaskControllerTest {
     // given
     final var request =
         new UserTaskUpdateRequest()
-            .changeset(new Changeset().candidateGroups(List.of()).followUpDate(""));
+            .changeset(
+                new UserTaskUpdateRequestChangeset().candidateGroups(List.of()).followUpDate(""));
 
     // when / then
     webClient
@@ -378,7 +382,7 @@ public class UserTaskControllerTest {
         new UserTaskUpdateRequest()
             .action("customAction")
             .changeset(
-                new Changeset()
+                new UserTaskUpdateRequestChangeset()
                     .addCandidateGroupsItem("foo")
                     .addCandidateUsersItem("bar")
                     .dueDate(TEST_TIME)
@@ -444,7 +448,8 @@ public class UserTaskControllerTest {
   @Test
   public void shouldYieldBadRequestWhenUpdateTaskWithoutMalformedDueDate() {
     // given
-    final var request = new UserTaskUpdateRequest().changeset(new Changeset().dueDate("foo"));
+    final var request =
+        new UserTaskUpdateRequest().changeset(new UserTaskUpdateRequestChangeset().dueDate("foo"));
 
     final var expectedBody =
         ProblemDetail.forStatusAndDetail(
@@ -474,7 +479,9 @@ public class UserTaskControllerTest {
   @Test
   public void shouldYieldBadRequestWhenUpdateTaskWithMalformedFollowUpDate() {
     // given
-    final var request = new UserTaskUpdateRequest().changeset(new Changeset().followUpDate("foo"));
+    final var request =
+        new UserTaskUpdateRequest()
+            .changeset(new UserTaskUpdateRequestChangeset().followUpDate("foo"));
 
     final var expectedBody =
         ProblemDetail.forStatusAndDetail(
@@ -505,7 +512,8 @@ public class UserTaskControllerTest {
   public void shouldYieldBadRequestWhenUpdateTaskWithoutMalformedFollowUpAndDueDate() {
     // given
     final var request =
-        new UserTaskUpdateRequest().changeset(new Changeset().dueDate("bar").followUpDate("foo"));
+        new UserTaskUpdateRequest()
+            .changeset(new UserTaskUpdateRequestChangeset().dueDate("bar").followUpDate("foo"));
 
     final var expectedBody =
         ProblemDetail.forStatusAndDetail(
@@ -536,9 +544,7 @@ public class UserTaskControllerTest {
   @Test
   public void shouldYieldBadRequestWhenUpdateTaskWithUntrackedChanges() {
     // given
-    final var request =
-        new UserTaskUpdateRequest()
-            .changeset(new Changeset().putAdditionalProperty("elementInstanceKey", 123456L));
+    final var request = new UserTaskUpdateRequest().changeset(new UserTaskUpdateRequestChangeset());
 
     final var expectedBody =
         ProblemDetail.forStatusAndDetail(

